src/System.Text.Json/tests/JsonReaderTests.cs

```diff
@@ -0,0 +1,88 @@
+﻿using System.Collections.Generic;
+using System.Text.Utf8;
+using Xunit;
+
+namespace System.Text.Json.Tests
+{
+    public class JsonReaderTests
+    {
+        [Fact]
+        public void ReadJsonUtf8()
+        {
+            Console.WriteLine(Json);
+            var str = new Utf8String(Json);
+            var jsonReader = new JsonReader(str);
+            var result = Read(jsonReader);
+            Assert.Equal(ExpectedAge, result.Age);
+            Assert.Equal(_expectedFullName, new Utf8String(result.FirstName + " " + result.LastName));
+            var phoneList = new List<Utf8String> {result.Phone1, result.Phone2};
+            Assert.Equal(ExpectedPhone, phoneList.ToArray());
+            Assert.Equal(_expectedAddress, new Utf8String(result.Street + ", " + result.City + ", " + result.Zip));
+            Assert.Equal(ExpectedId1, result.Id1);
+            Assert.Equal(ExpectedId2, result.Id2);
+        }
+
+        private const string Json =
+            "{\"age\":30,\"first\":\"John\",\"last\":\"Smith\",\"phoneNumbers\":[\"425-000-1212\",\"425-000-1213\"],\"address\":{\"street\":\"1 Microsoft Way\",\"city\":\"Redmond\",\"zip\":98052}\"IDs\" \n : [ \t 425 \n , 070 \n ] \t }";
+
+        private const uint ExpectedAge = 30;
+        private const uint ExpectedId1 = 425;
+        private const uint ExpectedId2 = 70;
+        private readonly Utf8String _expectedFullName = new Utf8String("John Smith");
+
+        private static readonly Utf8String[] ExpectedPhone =
+        {
+            new Utf8String("425-000-1212"),
+            new Utf8String("425-000-1213")
+        };
+
+        private readonly Utf8String _expectedAddress = new Utf8String("1 Microsoft Way, Redmond, 98052");
+
+        private static ReadObject Read(JsonReader json)
+        {
+            var output = new ReadObject();
+
+            json.ReadObjectStart();
```

Comment:
I assumed the user knew the json schema that he was trying to read. Should I enable the usability of the json reader as defined by @krwq?